tcms/testplans/static/testplans/js/get.js

```diff
@@ -0,0 +1,145 @@
+                                    <li><a href="#">Another Action</a></li>
+                                    <li><a href="#">Something Else Here</a></li>
+                                    <li role="separator" class="divider"></li>
+                                    <li><a href="#">Separated Link</a></li>
+                                </ul>
+                            </div>
+                        </div>
+                        <div class="list-view-pf-main-info">TC-${testCase.id}: ${testCase.summary}</div>
+                    </div>
+                    <div class="list-group-item-container container-fluid hidden">
+                        <div class="close">
+                            <span class="pficon pficon-close"></span>
+                        </div>
+                        <div class="row">
+                            <div class="col-md-12">Some test case widgets</div>
+                        </div>
+                    </div>
+                </div>`;
+    };
+
+    var addTestCasesAction = function() {
+        $("#confirmed-testcases input[type='checkbox']").change(function(e) {
+            if ($(this).is(":checked")) {
+                $(this).closest('.list-group-item').addClass("active");
+            } else {
+                $(this).closest('.list-group-item').removeClass("active");
+            }
+        });
+
+        // toggle dropdown menu
+        // $("#confirmed-testcases .list-view-pf-actions").on('show.bs.dropdown', function() {
+        //     var $this = $(this);
+        //     var $dropdown = $this.find('.dropdown');
+        //     var space = $(window).height() - $dropdown[0].getBoundingClientRect().top - $this.find('.dropdown-menu').outerHeight(true);
+        //     $dropdown.toggleClass('dropup', space < 10);
+        // });
+
+        // click the list-view heading then expand a row
+        $("#confirmed-testcases .list-group-item-header").click(function(event) {
+            if (!$(event.target).is("button, a, input, .fa-ellipsis-v")) {
+                $(this).find(".fa-angle-right").toggleClass("fa-angle-down")
+                    .end().parent().toggleClass("list-view-pf-expand-active")
+                    .find(".list-group-item-container").toggleClass("hidden");
+            }
+        });
+
+        // click the close button, hide the expand row and remove the active status
+        $("#confirmed-testcases .list-group-item-container .close").on("click", function() {
+            $(this).parent().addClass("hidden")
+                .parent().removeClass("list-view-pf-expand-active")
+                .find(".fa-angle-right").removeClass("fa-angle-down");
+        });
+    };
+
+    var getTestPlanId = function() {
+        var element = document.getElementById('test_plan_pk');
+        if (element && element.dataset && element.dataset.testplanPk) {
+            return element.dataset.testplanPk;
+        }
+
+        return null;
```

Comment:
Use jQuery for selection and pass the planId as argument to init() from .ready().
You can also use `$(selector).data('param')` and skip all the checks. Let it crash if something isn't found.

Reply:
I guess that inside the jquery same checks are done but with a little bit overhead. I want to avoid jquery as much as I can. I agree for the planId.

Reply:
We are using jQuery everywhere, the entire Patternfly is also build on top of it so there is no point in avoiding it.